cpp/include/milvus-storage/properties.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Zilliz
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <unordered_map>
+#include <string>
+
+namespace milvus_storage::api {
+using Properties = std::unordered_map<std::string, std::string>;
```

Comment:
should be `std::unordered_map<std::string, T>;` in the c++ side.

The public api can be `string key, string value`. or `const char *key, uint32_t keysz,const char *value, uint32_t valsz`

The `value` can cast to the `T` type, if it can't then return a `error type of arrow::status`.